pytorch_lightning/trainer/trainer.py

```diff
@@ -920,32 +918,34 @@ def __test_using_best_weights(self, ckpt_path, test_dataloaders):
             model.load_state_dict(ckpt['state_dict'])
 
         # attach dataloaders
-        if test_dataloaders is not None:
-            self.data_connector.attach_dataloaders(model, test_dataloaders=test_dataloaders)
+        if dataloaders is not None:
+            self.data_connector.attach_dataloaders(model, test_dataloaders=dataloaders)
+
+        if self.validating:
+            self.validated_ckpt_paath = ckpt_path
```

Comment:
```suggestion
            self.validated_ckpt_path = ckpt_path
```
oh oh! no test failing

Reply:
Good catch. This is not tested yet because it'll be when `trainer.validate` is actually implemented